common/data/src/main/java/org/thingsboard/server/common/data/device/profile/ProtoTransportPayloadConfiguration.java

```diff
@@ -0,0 +1,200 @@
+ * limitations under the License.
+ */
+package org.thingsboard.server.common.data.device.profile;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.github.os72.protobuf.dynamic.DynamicSchema;
+import com.github.os72.protobuf.dynamic.EnumDefinition;
+import com.github.os72.protobuf.dynamic.MessageDefinition;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import com.squareup.wire.schema.Location;
+import com.squareup.wire.schema.internal.parser.EnumConstantElement;
+import com.squareup.wire.schema.internal.parser.EnumElement;
+import com.squareup.wire.schema.internal.parser.FieldElement;
+import com.squareup.wire.schema.internal.parser.MessageElement;
+import com.squareup.wire.schema.internal.parser.OneOfElement;
+import com.squareup.wire.schema.internal.parser.ProtoFileElement;
+import com.squareup.wire.schema.internal.parser.ProtoParser;
+import com.squareup.wire.schema.internal.parser.TypeElement;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.thingsboard.server.common.data.TransportPayloadType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Data
+public class ProtoTransportPayloadConfiguration implements TransportPayloadTypeConfiguration {
+
+    public static final Location LOCATION = new Location("", "", -1, -1);
+    public static final String ATTRIBUTES_PROTO_SCHEMA = "attributes proto schema";
+    public static final String TELEMETRY_PROTO_SCHEMA = "telemetry proto schema";
+
+    private String deviceTelemetryProtoSchema;
+    private String deviceAttributesProtoSchema;
+
+    @Override
+    public TransportPayloadType getTransportPayloadType() {
+        return TransportPayloadType.PROTOBUF;
+    }
+
+    public Descriptors.Descriptor getTelemetryDynamicMessageDescriptor(String deviceTelemetryProtoSchema) {
+        return getDescriptor(deviceTelemetryProtoSchema, TELEMETRY_PROTO_SCHEMA);
+    }
+
+    public Descriptors.Descriptor getAttributesDynamicMessageDescriptor(String deviceAttributesProtoSchema) {
+        return getDescriptor(deviceAttributesProtoSchema, ATTRIBUTES_PROTO_SCHEMA);
+    }
+
+    private Descriptors.Descriptor getDescriptor(String protoSchema, String schemaName) {
+        try {
+            ProtoFileElement protoFileElement = getTransportProtoSchema(protoSchema);
+            DynamicSchema dynamicSchema = getDynamicSchema(protoFileElement, schemaName);
+            String lastMsgName = getMessageTypes(protoFileElement.getTypes()).stream()
+                    .map(MessageElement::getName).reduce((previous, last) -> last).get();
+            DynamicMessage.Builder builder = dynamicSchema.newMessageBuilder(lastMsgName);
+            return builder.getDescriptorForType();
+        } catch (Exception e) {
+            log.warn("Failed to get Message Descriptor due to {}", e.getMessage());
+            return null;
```

Comment:
throw new RuntimeException("Failed to get Message Descriptor due to :" + e.getMessage()); to be consistent with the getDynamicSchema